sources/packages/backend/libs/integrations/src/institution-integration/ece-integration/ece-response.processing.service.ts

```diff
@@ -18,38 +45,285 @@ export class ECEResponseProcessingService {
+    this.logger.log(`Starting download of file ${remoteFilePath}.`);
+    try {
+      const eceFileDetailRecords =
+        await this.integrationService.downloadResponseFile(remoteFilePath);
+      isECEResponseFileExist = !!eceFileDetailRecords.length;
+      // Check if the file exist in remote server and summary info
+      // if the file is not found.
+      if (!isECEResponseFileExist) {
+        const warningMessage = `File ${remoteFilePath} not found.`;
+        processSummary.summary.push(warningMessage);
+        this.logger.log(warningMessage);
+        return processSummary;
+      }
+      // Sanitize all the ece response detail records and
+      // transform them to disbursements which could be individually processed.
+      const disbursementsToProcess =
+        this.sanitizeAndTransformToDisbursements(eceFileDetailRecords);
+      const auditUser = await this.systemUsersService.systemUser();
+      const {
+        totalDisbursements,
+        disbursementsSuccessfullyProcessed,
+        disbursementsSkipped,
+        duplicateDisbursements,
+        disbursementsFailedToProcess,
+      } = await this.validateAndUpdateEnrolmentStatus(
+        disbursementsToProcess,
+        auditUser.id,
+        processSummary,
+      );
+      processSummary.summary.push(
+        `Total disbursements found: ${totalDisbursements}`,
+      );
+      processSummary.summary.push(
+        `Disbursements successfully updated: ${disbursementsSuccessfullyProcessed}`,
+      );
+      processSummary.summary.push(
+        `Disbursements skipped to be processed: ${disbursementsSkipped}`,
+      );
+      processSummary.summary.push(
+        `Disbursements considered duplicate and skipped: ${duplicateDisbursements}`,
+      );
+      processSummary.summary.push(
+        `Disbursements failed to process: ${disbursementsFailedToProcess}`,
+      );
+      this.logger.log(`SCompleted processing the file ${remoteFilePath}.`);
+    } catch (error: unknown) {
+      this.logger.error(error);
+      processSummary.errors.push(
+        `Error processing the file ${remoteFilePath}. ${error}`,
+      );
+    } finally {
+      // Delete the ECE response file, if the file exist in remote server.
+      if (isECEResponseFileExist) {
+        await this.deleteTheProcessedFile(remoteFilePath, processSummary);
+      }
+    }
+    return processSummary;
+  }
+
+  /**
+   * Sanitize the the detail records and transform the
+   * detail records to individual disbursements.
+   * @param eceFileDetailRecords detail records of ece file.
+   * @returns disbursements to be processed.
+   */
+  private sanitizeAndTransformToDisbursements(
+    eceFileDetailRecords: ECEResponseFileDetail[],
+  ): ECEDisbursements {
+    const disbursements = {} as ECEDisbursements;
+    for (const eceDetailRecord of eceFileDetailRecords) {
+      const errorMessage = eceDetailRecord.getInvalidDataMessage();
```

Comment:
Once we get any results from the `eceDetailRecord.getInvalidDataMessage();` we can avoid processing the file itself but I believe that we can have the method `getInvalidDataMessage` executed for all the records.